rust/feature-flags/src/flag_matching.rs

```diff
@@ -579,6 +598,23 @@ impl FeatureFlagMatcher {
         }
         Ok(None)
     }
+
+    fn get_matching_payload(
+        &self,
+        is_match: bool,
+        match_variant: Option<&str>,
+        feature_flag: &FeatureFlag,
+    ) -> Option<serde_json::Value> {
+        if is_match {
+            if let Some(variant) = match_variant {
```

Comment:
You could make use of `match_variant.unwrap_or("true")` here to avoid some nesting. Also, the `is_match` part feels weird - if the caller is passing you a bool, can't they just use that bool to decide whether to call this or not?

Reply:
yeah you're right, I was probably too defensive here for this method; I can just call it from within the parent method `is_match` block and then avoid the bool all together.